### PR TITLE
Emit error event for errors handled by an error boundary

### PR DESCRIPTION
**Pull request by sventschui**

### CODE REVIEW

**Review by JoviDeCroock — APPROVED** [review]
Nice catch!

**Review by marvinhagemeister — APPROVED** [review]
Awesome find! :raised_hands:

File: debug/src/debug.js [hunk 1]
```diff
@@ -67,7 +67,25 @@ export function initDebug() {
...
+			// and thus make the Next.js dev overlay work.
+			if (typeof error.then != 'function') {
+				const errorEvent = new ErrorEvent(error.message, {
+					error,
+					message: error.message,
+					lineno: error.lineno,
+					filename: error.filename
+				});
+				errorEvent.initEvent('error', true, true);
+				window.dispatchEvent(errorEvent);
```
Comment (developit) [inline debug/src/debug.js:84]: I believe this accomplishes the same:

```suggestion
			setTimeout(() => { throw error; });
```

File: debug/src/debug.js [hunk 2]
```diff
@@ -67,7 +67,21 @@ export function initDebug() {
...
-		oldCatchError(error, vnode, oldVNode);
+		try {
+			oldCatchError(error, vnode, oldVNode);
+
+			// when an error was handled by an ErrorBoundary we will nontheless emit an error
+			// event on the window object. This is to make up for react compatibility in dev mode
+			// and thus make the Next.js dev overlay work.
+			if (typeof error.then != 'function') {
+				setTimeout(() => {
+					console.log('re-throw', error);
```
Comment (marvinhagemeister) [inline debug/src/debug.js:78]: FYI: This was probable left in by accident.

Comment (developit) [reply debug/src/debug.js:78]:
```suggestion
```

File: debug/src/debug.js [hunk 3]
```diff
@@ -67,7 +67,21 @@ export function initDebug() {
...
+					throw error;
+				}, 0);
```
Comment (developit) [inline debug/src/debug.js:80]:
```suggestion
				});
```

**Review by developit — APPROVED** [review]

**Pull request merged**